homeassistant/components/tradfri/cover.py

```diff
@@ -26,9 +26,13 @@ async def async_setup_entry(
     api = tradfri_data[KEY_API]
     devices = tradfri_data[DEVICES]
 
-    covers = [dev for dev in devices if dev.has_blind_control]
-    if covers:
-        async_add_entities(TradfriCover(cover, api, gateway_id) for cover in covers)
+    entities = [
+        TradfriCover(dev, api, gateway_id) for dev in devices if dev.has_blind_control
+    ]
+    if len(entities) > 0:
+        async_add_entities(entities)
+        for entity in entities:
+            entity.setup()
```

Comment:
We don't need to change any of the platforms.

Reply:
Nope.